synapse/storage/search.py

```diff
@@ -252,12 +252,17 @@ def search_room(self, room_id, search_term, keys, limit, pagination_token=None):
                 " WHERE vector @@ query AND room_id = ?"
             )
         elif isinstance(self.database_engine, Sqlite3Engine):
+            # We use CROSS JOIN here to ensure we use the right indexes.
```

Comment:
It might be worth elaborating on what the right indexes are. Something like "We want to use the full text search index on event_search to extract all possible matches first, then lookup those matches in the events table to get the topological ordering. We need to use the indexes in this order because sqlite refuses to MATCH unless it uses the full text search index"